src/store.ts

```diff
@@ -93,16 +103,42 @@ export class GraphDbDataStore implements Store {
       .toArray()[0];
 
     const datasetUri = quad.subject.value;
-    const writer = new Writer({format: 'application/trig'});
-    writer.addQuads(dataset.toArray());
-    await writer.end(async (error, result) => {
+
+    await getWriter(dataset.toArray()).end(async (error, result) => {
       await this.request(
         'PUT',
         '/rdf-graphs/service?graph=' + datasetUri,
         result
       );
     });
 
+    await this.request(
+      'DELETE',
+      '/statements?' +
+        querystring.stringify({
+          subj: '<' + url.toString() + '>',
+          context: '<' + this.registrationsGraph + '>',
+        })
+    );
+
+    const foundAtUrlQuads = [
+      factory.quad(
+        factory.namedNode(url.toString()),
+        factory.namedNode('http://schema.org/mainEntity'),
+        factory.namedNode(datasetUri),
+        factory.namedNode(this.registrationsGraph)
+      ),
+      factory.quad(
+        factory.namedNode(url.toString()),
+        factory.namedNode('http://schema.org/datePosted'),
```

Comment:
@coret I use `schema:mainEntity` and `schema:datePosted` (for the date at which the URL was last registered). Do these predicates make sense to you?

Reply:
I have not found a vocabulary which has all the properties we need (as defined in 'Registered URL’s of dataset descriptions and catalogs' of the Design). As this data is for internal use only, I think we can define our own vocabulary like:
```
<https://www.openarch.nl/datasets/hga> a nde:CrawlResource
	nde:status		"valid"				(or invalid, deleted)
	nde:dateAdded 		....(DateTime)
	nde:dateLastFetch 	....(DateTime)
	nde:resourceType  	"dataset"  			(or "datasets", "datacatalog")
	nde:encodingFormat	"inlined-json-ld"		(or "microdata", "json-ld", "turtle", "nt", "rdf-xml")
	nde:foundVocabulary	"schema"			(or "dcat")
	nde:datasetsExtracted	<https://www.openarch.nl/id/dataset_hga>
```
When a CrawlResource, specified by @id (=url), already exists, the existing CrawlResource should not be updated nor should a new one be created.

Reply:
It’s good practice to prefer re-using existing ontologies over inventing your own.

In this case, I agree that Schema.org doesn’t fully match what we want. However, you have some redundant properties that can also change between crawls and therefore need to be checked each time anyway (so don’t need to be stored): 
`nde:resourceType` (we store datasets only), `nde:encodingFormat` and `nde:foundVocabulary` (only interesting for internal statistics). 

We’re then left with some properties for which Schema ‘equivalents’ do exist:

- `nde:dateLastFetch` = `schema:dateRead`
- `nde:dateAdded` = `schema:datePosted`
- `nde:datasetsExtracted` = `schema:mainEntity` or `schema:about`.

Reply:
`https://schema.org/status`